ui/src/main/webapp/components/FileInputComponent.tsx

```diff
@@ -63,14 +92,14 @@ function FileInputComponent(props) {
                 if (isValid === true) {
                     setErrorMsg('');
                     try {
-                        handleChange(field, textDecoder.decode(fileReader.result));
+                        handleChange(field, textDecoder.decode(fileReader.result as ArrayBuffer));
```

Comment:
please avoid casting, narrow down the type instead. You have union here: `string | ArrayBuffer | null`. Add checks to make sure we have ArrayBuffer, not string or null.

Reply:
done